src/traits.rs

```diff
@@ -299,7 +299,7 @@ pub trait Random: Sized {
     /// Generate a random value.
     ///
     /// If `rng` is a CSRNG, the generation is cryptographically secure as well.
-    fn random<R: RngCore + ?Sized>(rng: &mut R) -> Self;
+    fn random<R: RngCore>(rng: &mut R) -> Self;
```

Comment:
Please also add a `try_random` method to this trait. I think you also could add a blanket impl for `random` in terms of `try_random`. Same for `random_mod`/`try_random_mod`.